dip-template/runtimes/dip-consumer/src/xcm_config.rs

```diff
@@ -39,12 +39,13 @@ use crate::{
 
 parameter_types! {
 	pub HereLocation: MultiLocation = MultiLocation::here();
+	pub NoneNetworkId: Option<NetworkId> = None;
 	pub UnitWeightCost: Weight = Weight::from_ref_time(1_000);
 	pub UniversalLocation: InteriorMultiLocation = Parachain(ParachainInfo::parachain_id().into()).into();
 }
 
-pub type Barrier = AllowTopLevelPaidExecutionFrom<Everything>;
-pub type AssetTransactorLocationConverter = SiblingParachainConvertsVia<Sibling, AccountId>;
+pub type Barrier = OkOrElseCheckForParachainProvider<AllowTopLevelPaidExecutionFrom<Nothing>, ConstU32<2_000>>;
```

Comment:
Can this be accomplished with this?

```
type Barrier = WithComputedOrigin<(AllowTopLevelPaidExecutionFrom<DerivativeCustomerLocations>);
```

Reply:
As we discussed, the design at the time of this comment was flawed. I have now added a `ExpectOrigin` instruction early on (https://github.com/KILTprotocol/kilt-node/pull/528/commits/d683a3f67cbc4e7ad1e99d39a4fa441a61fbf807). For now, this is a sufficiently good approach, which means that the `WithComputedOrigin` barrier would not let the call through, which is not what we want. Also the `AllowTopLevelPaidExecutionFrom` would fail. There is no way around using the `OkOrElseCheckForParachainProvider` barrier provided here. If some barrier must explicitly fail, the `ErrOrElseCheckForParachainProvider` should be used.